blueox/network.py

```diff
@@ -80,6 +80,11 @@ def _serialize_context(context):
         if len(context_dict.get(key, "")) > 64:
             raise ValueError("Value too long: %r" % key)
 
+    # type might be unicode, host and end less likely
+    type = context_dict['type']
+    if isinstance(type, unicode):
+        type = unicodedata.normalize('NFKD', type).encode('ascii','ignore')
```

Comment:
this will drop all non-ascii characters. is that really the best option? if you just need `type` to be `str` instead of `unicode` you should just do `type.encode('utf-8')`.
and you should really do this for host and end too. how about this:
```
context_dict = {k: v.encode('utf-8') if isinstance(v, unicode) else v for k, v in context_dict.items()
```
msgpack treats `str` and `unicode` identically anyway so this will have no effect for anything not in meta_data.

Reply:
ufaaa